File: src/main/java/client/effects/field/ScreenFieldEffect.java
package client.effects.field;

import client.effects.AbstractFieldEffect;
import client.effects.FieldEffectType;
import util.packet.PacketWriter;

public class ScreenFieldEffect extends AbstractFieldEffect {

    private final String path;

    public ScreenFieldEffect(String path) {
        this.path = path;
    }

    @Override
    public FieldEffectType getType() {
        return FieldEffectType.SCREEN;
    }

    @Override
    protected void encodeData(PacketWriter pw) {
        pw.writeMapleString(path);
    }
}

File: src/main/java/client/effects/field/ScreenFieldEffect.kt
package client.effects.field

import client.effects.AbstractFieldEffect
import client.effects.FieldEffectType
import util.packet.PacketWriter

class ScreenFieldEffect(val path: String) : AbstractFieldEffect() {

    override val type = FieldEffectType.SCREEN

    override fun encodeData(pw: PacketWriter) {
        pw.writeMapleString(path)
    }
}